site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```diff
@@ -9163,20 +9163,23 @@ private void actionForTemplate(String direction, int index,
 							final String userEmail = UserDirectoryService.getCurrentUser().getEmail();
 							final Session session = SessionManager.getCurrentSession();
 							final ToolSession toolSession = SessionManager.getCurrentToolSession();
+							final String siteId = existingSite.getId();
 							Thread siteImportThread = new Thread(){
 								public void run() {
-									Site existingSite = getStateSite(state);
-									EventTrackingService.post(EventTrackingService.newEvent(SiteService.EVENT_SITE_IMPORT_START, existingSite.getReference(), false));
-									SessionManager.setCurrentSession(session);
-									SessionManager.setCurrentToolSession(toolSession);
-									importToolIntoSite(existingTools, importTools, existingSite);
-									existingSite = getStateSite(state); // refresh site for
-									// WC and News
-									commitSite(existingSite);
```

Comment:
The comitSite was really just for the UX. If you imported a Lesson, then commitSite would make sure that it refreshed the tools list and showed up on the left hand side. It's not needed since this is done in its own thread.